sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2033,62 +2036,11 @@ class Analyzer(
    */
   object ResolveAggregateFunctions extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
-      case f @ Filter(cond, agg @ Aggregate(grouping, originalAggExprs, child)) if agg.resolved =>
+      case AggregateWithHaving(cond, agg: Aggregate) if agg.resolved =>
```

Comment:
Could you leave some comments here about why we need this special handling for aggregate with having?

Reply:
Sure, done in c75fe08.